### ISSUE TITLE
Can't use activity outputs in mappings

### ISSUE THREAD

**Issue by fcastill** [TIBCOSoftware/flogo#176]
I'm trying to use the new model and I'm not able to use a task output as an input of another task. I get an error `failed to resolve activity attr: 'message', not found in flow`.

What I'm doing is:

1. Using the app provided below create a new flogo project: `flogo create -flv github.com/TIBCOSoftware/flogo-contrib/activity/log@master,github.com/TIBCOSoftware/flogo-lib/app/resource@master -f "map-test.json" map-test`
2. Inside created map-test directory I run `flogo build`
3. Insided map-test/bin run  `FLOGO_LOG_LEVEL=DEBUG  ./map-test``
4. GET request to `localhost:8585/test`

This is the app I'm using:

```javascript
{
  "name": "map-test-2",
  "type": "flogo:app",
  "version": "0.0.1",
  "appModel": "1.0.0",
  "triggers": [
    {
      "id": "receive_http_message",
      "ref": "github.com/TIBCOSoftware/flogo-contrib/trigger/rest",
      "name": "Receive HTTP Message",
      "description": "Simple REST Trigger",
      "settings": {
        "port": "8585"
      },
      "handlers": [
        {
          "action": {
            "ref": "github.com/TIBCOSoftware/flogo-contrib/action/flow",
            "data": {
              "flowURI": "res://flow:simple_map"
            }
          },
          "settings": {
            "method": "GET",
            "path": "/test"
          }
        }
      ]
    }
  ],
  "resources": [
    {
      "id": "flow:simple_map",
      "data": {
        "name": "Simple map",
        "tasks": [
          {
            "id": "log_2",
            "activity": {
              "ref": "github.com/TIBCOSoftware/flogo-contrib/activity/log",
              "input": {
                "message": "From log 2",
                "flowInfo": "false",
                "addToFlow": "true"
              }
            }
          },
          {
            "id": "log_3",
            "activity": {
              "ref": "github.com/TIBCOSoftware/flogo-contrib/activity/log",
              "input": {
                "message": "",
                "flowInfo": "true",
                "addToFlow": "false"
              },
              "mappings": {
                "input": [
                  {
                    "type": "assign",
                    "value": "$activity[log_2].message",
                    "mapTo": "message"
                  }
                ]
              }
            }
          }
        ],
        "links": [
          {
            "from": "log_2",
            "to": "log_3"
          }
        ]
      }
    }
  ]
}
```

These are the logs:

```
2018-03-05 18:39:21.295 DEBUG  [engine] - Registering activity: 'github.com/TIBCOSoftware/flogo-contrib/activity/log'
2018-03-05 18:39:21.295 INFO   [engine] - Engine: Starting...
2018-03-05 18:39:21.296 WARN   [engine] - Unsupported link type '%!s(MISSING)', using default link
2018-03-05 18:39:21.296 DEBUG  [trigger-tibco-rest] - Registering handler [GET: /test]
2018-03-05 18:39:21.296 DEBUG  [trigger-tibco-rest] - Configured on port 8585
2018-03-05 18:39:21.296 DEBUG  [engine] - ActionRunner Service: Starting...
2018-03-05 18:39:21.296 DEBUG  [engine] - Starting worker with id '1'
2018-03-05 18:39:21.296 DEBUG  [engine] - Starting worker with id '2'
2018-03-05 18:39:21.296 DEBUG  [engine] - Starting worker with id '3'
2018-03-05 18:39:21.296 DEBUG  [engine] - Starting worker with id '4'
2018-03-05 18:39:21.296 DEBUG  [engine] - Starting worker with id '5'
2018-03-05 18:39:21.296 DEBUG  [engine] - ActionRunner Service: Started
2018-03-05 18:39:21.296 INFO   [engine] - Engine: Starting Services...
2018-03-05 18:39:21.296 INFO   [engine] - Engine: Started Services
2018-03-05 18:39:21.296 INFO   [engine] - Engine: Starting Triggers...
2018-03-05 18:39:21.296 DEBUG  [engine] - Trigger [ receive_http_message ]: Starting...
2018-03-05 18:39:21.297 DEBUG  [engine] - Trigger [ receive_http_message ]: Started
2018-03-05 18:39:21.297 INFO   [engine] - Trigger [ receive_http_message ]: Started
2018-03-05 18:39:21.297 INFO   [engine] - Engine: Triggers Started
2018-03-05 18:39:21.297 INFO   [engine] - Engine: Started
2018-03-05 18:39:28.729 INFO   [trigger-tibco-rest] - Received request for id 'receive_http_message'
2018-03-05 18:39:28.729 DEBUG  [engine] - iomd (*data.IOMetadata)(nil)
2018-03-05 18:39:28.729 DEBUG  [engine] - No mapping specified, adding trigger outputs as inputs to action
2018-03-05 18:39:28.729 DEBUG  [engine] -  Attr: params, Type: params, Value: map[]
2018-03-05 18:39:28.729 DEBUG  [engine] -  Attr: pathParams, Type: params, Value: map[]
2018-03-05 18:39:28.729 DEBUG  [engine] -  Attr: queryParams, Type: params, Value: map[]
2018-03-05 18:39:28.729 DEBUG  [engine] -  Attr: header, Type: params, Value: map[Content-Type:application/json Cache-Control:no-cache Postman-Token:fbe0207e-4272-443f-aa20-54ba45355031 User-Agent:PostmanRuntime/7.1.1 Accept:*/* Accept-Encoding:gzip, deflate Connection:keep-alive]
2018-03-05 18:39:28.729 DEBUG  [engine] -  Attr: content, Type: object, Value: map[]
2018-03-05 18:39:28.729 DEBUG  [engine] - Action 'github.com/TIBCOSoftware/flogo-contrib/action/flow' queued
2018-03-05 18:39:28.729 DEBUG  [engine] - Received work request
2018-03-05 18:39:28.729 DEBUG  [engine] - Dispatching work request
2018-03-05 18:39:28.729 DEBUG  [engine] - Action-Worker-1: Received Request
2018-03-05 18:39:28.729 INFO   [engine] - Running FlowAction for URI: 'res://flow:simple_map'
2018-03-05 18:39:28.729 DEBUG  [engine] - Creating Flow Instance: ec82090a55f7069b6ba5f7a5ac1d6419
2018-03-05 18:39:28.729 DEBUG  [engine] - Executing Flow Instance: ec82090a55f7069b6ba5f7a5ac1d6419
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Enter Task 'log_2'
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Task 'log_2' Ready
2018-03-05 18:39:28.729 DEBUG  [engine] - Scheduling task 'log_2'
2018-03-05 18:39:28.729 DEBUG  [engine] - Step: 1
2018-03-05 18:39:28.729 DEBUG  [engine] - Retrieved item from Flow Instance work queue
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Eval Task 'log_2'
2018-03-05 18:39:28.729 INFO   [activity-flogo-log] - From log 2
2018-03-05 18:39:28.729 DEBUG  [engine] - SET OUTPUT: message = From log 2

2018-03-05 18:39:28.729 DEBUG  [engine] - SetAttr: message = From log 2
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Task 'log_2' is done
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Task 'log_2' has 1 outgoing links
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Task 'log_2': Following Link  to task 'log_3'
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Enter Task 'log_3'
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Task 'log_3' has 1 incoming Links
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Task 'log_3': Link from Task 'log_2' has status 'true'
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Task 'log_3' Ready
2018-03-05 18:39:28.729 DEBUG  [engine] - Scheduling task 'log_3'
2018-03-05 18:39:28.729 DEBUG  [engine] - Step: 2
2018-03-05 18:39:28.729 DEBUG  [engine] - Retrieved item from Flow Instance work queue
2018-03-05 18:39:28.729 DEBUG  [flowmodel-simple] - Eval Task 'log_3'
2018-03-05 18:39:28.729 DEBUG  [engine] - Applying InputMapper
2018-03-05 18:39:28.729 DEBUG  [engine] - Action-Worker-1: Received result: &runner.ActionResult{results:map[string]*data.Attribute{"id":(*data.Attribute)(0xc4201849c0)}, err:error(nil)}
2018-03-05 18:39:28.729 DEBUG  [basic-mapper] - Updated mapping def &{Type:1 Value:$activity[log_2].message MapTo:message}
2018-03-05 18:39:28.729 DEBUG  [engine] - Action 'github.com/TIBCOSoftware/flogo-contrib/action/flow' returned
2018-03-05 18:39:28.729 ERROR  [engine] - Execution failed for Activity[] in Flow[Simple map] - failed to resolve activity attr: 'message', not found in flow
2018-03-05 18:39:28.729 ERROR  [flowmodel-simple] - Error evaluating activity 'log_3'[github.com/TIBCOSoftware/flogo-contrib/activity/log] - failed to resolve activity attr: 'message', not found in flow
2018-03-05 18:39:28.729 DEBUG  [engine] - AddAttr - name: {Error.activity}, type: string, value:
2018-03-05 18:39:28.729 DEBUG  [engine] - AddAttr - name: {Error.message}, type: string, value:failed to resolve activity attr: 'message', not found in flow
2018-03-05 18:39:28.729 DEBUG  [engine] - AddAttr - name: {Error.type}, type: string, value:mapper
2018-03-05 18:39:28.729 DEBUG  [engine] - Done Executing flow instance [ec82090a55f7069b6ba5f7a5ac1d6419] - Status: 700
2018-03-05 18:39:28.729 INFO   [engine] - Flow instance [ec82090a55f7069b6ba5f7a5ac1d6419] Failed
2018-03-05 18:39:28.729 DEBUG  [engine] - Action-Worker-1: Completed Request
```

**Comment by fcastill** [comment]
It works with the legacy model:

```
{
  "name": "map-test-legacy-7",
  "type": "flogo:app",
  "version": "0.0.1",
  "description": "",
  "triggers": [
    {
      "name": "Receive HTTP Message",
      "ref": "github.com/TIBCOSoftware/flogo-contrib/trigger/rest",
      "description": "Simple REST Trigger",
      "settings": {
        "port": "8585"
      },
      "id": "receive_http_message",
      "handlers": [
        {
          "settings": {
            "method": "GET",
            "path": "/test"
          },
          "actionId": "simple_map",
          "actionMappings": {
            "input": [],
            "output": []
          }
        }
      ]
    }
  ],
  "actions": [
    {
      "id": "simple_map",
      "metadata": {
        "input": [],
        "output": []
      },
      "description": "",
      "data": {
        "flow": {
          "name": "Simple map",
          "type": 1,
          "attributes": [],
          "rootTask": {
            "id": "root",
            "type": 1,
            "tasks": [
              {
                "id": "log_2",
                "name": "Log 2",
                "description": "Simple Log Activity",
                "type": 1,
                "activityType": "github-com-tibco-software-flogo-contrib-activity-log",
                "activityRef": "github.com/TIBCOSoftware/flogo-contrib/activity/log",
                "attributes": [
                  {
                    "name": "message",
                    "value": "From log 2",
                    "required": false,
                    "type": "string"
                  },
                  {
                    "name": "flowInfo",
                    "value": "false",
                    "required": false,
                    "type": "boolean"
                  },
                  {
                    "name": "addToFlow",
                    "value": "true",
                    "required": false,
                    "type": "boolean"
                  }
                ]
              },
              {
                "id": "log_3",
                "name": "Log 3",
                "description": "Simple Log Activity",
                "type": 1,
                "activityType": "github-com-tibco-software-flogo-contrib-activity-log",
                "activityRef": "github.com/TIBCOSoftware/flogo-contrib/activity/log",
                "attributes": [
                  {
                    "name": "message",
                    "value": "",
                    "required": false,
                    "type": "string"
                  },
                  {
                    "name": "flowInfo",
                    "value": "true",
                    "required": false,
                    "type": "boolean"
                  },
                  {
                    "name": "addToFlow",
                    "value": "true",
                    "required": false,
                    "type": "boolean"
                  }
                ],
                "inputMappings": [
                  {
                    "type": 1,
                    "value": "$activity[log_2].message",
                    "mapTo": "message"
                  }
                ]
              }
            ],
            "links": [
              {
                "id": 1,
                "from": "log_2",
                "to": "log_3",
                "type": 0
              }
            ],
            "attributes": []
          }
        }
      }
    }
  ]
}
```

```
2018-03-05 18:52:29.341 DEBUG  [engine] - Registering activity: 'github.com/TIBCOSoftware/flogo-contrib/activity/log'
2018-03-05 18:52:29.342 INFO   [engine] - Engine: Starting...
2018-03-05 18:52:29.343 DEBUG  [trigger-tibco-rest] - Registering handler [GET: /test]
2018-03-05 18:52:29.343 DEBUG  [trigger-tibco-rest] - Configured on port 8585
2018-03-05 18:52:29.343 DEBUG  [engine] - ActionRunner Service: Starting...
2018-03-05 18:52:29.343 DEBUG  [engine] - Starting worker with id '1'
2018-03-05 18:52:29.343 DEBUG  [engine] - Starting worker with id '2'
2018-03-05 18:52:29.343 DEBUG  [engine] - Starting worker with id '3'
2018-03-05 18:52:29.343 DEBUG  [engine] - Starting worker with id '4'
2018-03-05 18:52:29.343 DEBUG  [engine] - Starting worker with id '5'
2018-03-05 18:52:29.343 DEBUG  [engine] - ActionRunner Service: Started
2018-03-05 18:52:29.343 INFO   [engine] - Engine: Starting Services...
2018-03-05 18:52:29.343 INFO   [engine] - Engine: Started Services
2018-03-05 18:52:29.343 INFO   [engine] - Engine: Starting Triggers...
2018-03-05 18:52:29.343 DEBUG  [engine] - Trigger [ receive_http_message ]: Starting...
2018-03-05 18:52:29.344 DEBUG  [engine] - Trigger [ receive_http_message ]: Started
2018-03-05 18:52:29.344 INFO   [engine] - Trigger [ receive_http_message ]: Started
2018-03-05 18:52:29.344 INFO   [engine] - Engine: Triggers Started
2018-03-05 18:52:29.344 INFO   [engine] - Engine: Started
2018-03-05 18:52:37.132 INFO   [trigger-tibco-rest] - Received request for id 'receive_http_message'
2018-03-05 18:52:37.132 DEBUG  [engine] - iomd &data.IOMetadata{Input:map[string]*data.Attribute{}, Output:map[string]*data.Attribute{}}
2018-03-05 18:52:37.132 DEBUG  [engine] - No mapping specified, adding trigger outputs as inputs to action
2018-03-05 18:52:37.132 DEBUG  [engine] -  Attr: queryParams, Type: params, Value: map[]
2018-03-05 18:52:37.133 DEBUG  [engine] -  Attr: header, Type: params, Value: map[Accept:*/* Accept-Encoding:gzip, deflate Connection:keep-alive Content-Type:application/json Cache-Control:no-cache Postman-Token:ffea84c3-89a9-4eaa-9f3d-5cdd138bbf22 User-Agent:PostmanRuntime/7.1.1]
2018-03-05 18:52:37.133 DEBUG  [engine] -  Attr: content, Type: object, Value: map[]
2018-03-05 18:52:37.133 DEBUG  [engine] -  Attr: params, Type: params, Value: map[]
2018-03-05 18:52:37.133 DEBUG  [engine] -  Attr: pathParams, Type: params, Value: map[]
2018-03-05 18:52:37.133 DEBUG  [engine] - Action 'github.com/TIBCOSoftware/flogo-contrib/action/flow' queued
2018-03-05 18:52:37.133 DEBUG  [engine] - Received work request
2018-03-05 18:52:37.133 DEBUG  [engine] - Dispatching work request
2018-03-05 18:52:37.133 DEBUG  [engine] - Action-Worker-1: Received Request
2018-03-05 18:52:37.133 INFO   [engine] - Running FlowAction for URI: 'res://flow:342811347'
2018-03-05 18:52:37.133 DEBUG  [engine] - Creating Flow Instance: e29f9d5b2e85429d5f66ef4bc1f1d5a7
2018-03-05 18:52:37.133 DEBUG  [engine] - Executing Flow Instance: e29f9d5b2e85429d5f66ef4bc1f1d5a7
2018-03-05 18:52:37.133 DEBUG  [flowmodel-simple] - Enter Task 'log_2'
2018-03-05 18:52:37.133 DEBUG  [flowmodel-simple] - Task 'log_2' Ready
2018-03-05 18:52:37.133 DEBUG  [engine] - Scheduling task 'log_2'
2018-03-05 18:52:37.133 DEBUG  [engine] - Step: 1
2018-03-05 18:52:37.133 DEBUG  [engine] - Retrieved item from Flow Instance work queue
2018-03-05 18:52:37.133 DEBUG  [flowmodel-simple] - Eval Task 'log_2'
2018-03-05 18:52:37.134 INFO   [activity-flogo-log] - From log 2
2018-03-05 18:52:37.134 DEBUG  [engine] - SET OUTPUT: message = From log 2

2018-03-05 18:52:37.134 DEBUG  [engine] - SetAttr: message = From log 2
2018-03-05 18:52:37.134 DEBUG  [engine] - Applying OutputMapper
2018-03-05 18:52:37.134 DEBUG  [engine] - Action-Worker-1: Received result: &runner.ActionResult{results:map[string]*data.Attribute{"id":(*data.Attribute)(0xc4202081e0)}, err:error(nil)}
2018-03-05 18:52:37.134 DEBUG  [engine] - Action 'github.com/TIBCOSoftware/flogo-contrib/action/flow' returned
2018-03-05 18:52:37.134 DEBUG  [engine] - AddAttr - name: _A.log_2.message, type: string, value:From log 2
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Task 'log_2' is done
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Task 'log_2' has 1 outgoing links
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Task 'log_2': Following Link  to task 'log_3'
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Enter Task 'log_3'
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Task 'log_3' has 1 incoming Links
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Task 'log_3': Link from Task 'log_2' has status 'true'
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Task 'log_3' Ready
2018-03-05 18:52:37.134 DEBUG  [engine] - Scheduling task 'log_3'
2018-03-05 18:52:37.134 DEBUG  [engine] - Step: 2
2018-03-05 18:52:37.134 DEBUG  [engine] - Retrieved item from Flow Instance work queue
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Eval Task 'log_3'
2018-03-05 18:52:37.134 DEBUG  [engine] - Applying InputMapper
2018-03-05 18:52:37.134 DEBUG  [basic-mapper] - Updated mapping def &{Type:1 Value:$activity[log_2].message MapTo:message}
2018-03-05 18:52:37.134 DEBUG  [engine] - SetAttr: message = From log 2
2018-03-05 18:52:37.134 INFO   [activity-flogo-log] - 'From log 2' - FlowInstanceID [e29f9d5b2e85429d5f66ef4bc1f1d5a7], Flow [Simple map], Task [Log 3]
2018-03-05 18:52:37.134 DEBUG  [engine] - SET OUTPUT: message = 'From log 2' - FlowInstanceID [e29f9d5b2e85429d5f66ef4bc1f1d5a7], Flow [Simple map], Task [Log 3]

2018-03-05 18:52:37.134 DEBUG  [engine] - SetAttr: message = 'From log 2' - FlowInstanceID [e29f9d5b2e85429d5f66ef4bc1f1d5a7], Flow [Simple map], Task [Log 3]
2018-03-05 18:52:37.134 DEBUG  [engine] - Applying OutputMapper
2018-03-05 18:52:37.134 DEBUG  [engine] - AddAttr - name: _A.log_3.message, type: string, value:'From log 2' - FlowInstanceID [e29f9d5b2e85429d5f66ef4bc1f1d5a7], Flow [Simple map], Task [Log 3]
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Task 'log_3' is done
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Notifying flow that end task 'log_3' is done
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Checking if all tasks done or skipped
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - All tasks done or skipped
2018-03-05 18:52:37.134 DEBUG  [flowmodel-simple] - Flow Done
2018-03-05 18:52:37.134 DEBUG  [engine] - Done Executing flow instance [e29f9d5b2e85429d5f66ef4bc1f1d5a7] - Status: 500
2018-03-05 18:52:37.134 INFO   [engine] - Flow instance [e29f9d5b2e85429d5f66ef4bc1f1d5a7] Completed Successfully
2018-03-05 18:52:37.134 DEBUG  [engine] - Action-Worker-1: Received result: &runner.ActionResult{results:map[string]*data.Attribute(nil), err:error(nil)}
2018-03-05 18:52:37.134 DEBUG  [engine] - Action-Worker-1: Completed Request
```

**Comment by mellistibco** [comment]
@fcastill is this still an issue since @lixingwang merged in TIBCOSoftware/flogo-contrib#218?

**Comment by fcastill** [comment]
Not an issue anymore.
